app/models/llms/basic_workflow/integration.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python3
+# Standard library
+import os
+import time
+import json
+import datetime
+from typing import Dict
+from enum import Enum
+from contextlib import contextmanager
+
+# Third-party libraries
+from openai import OpenAI
+from dotenv import load_dotenv
+
+@contextmanager
+def timer(operation_name: str):
+    start_time = time.time()
+    try:
+        yield
+    finally:
+        end_time = time.time()
+        execution_time = end_time - start_time
+        print(f"{operation_name}: {execution_time:.2f} seconds")
+
+
+class Budget(Enum):
+    BUDGET_FRIENDLY = "Budget-friendly"
+    MODERATE = "Moderate"
+    LUXURY = "Luxury"
+
+
+class TravelStyle(Enum):
+    FAMILY_FRIENDLY = "Family-friendly"
+    ADVENTURE = "Adventure"
+    CULTURAL = "Cultural"
+
+load_dotenv()
+
+class ItineraryGenerator:  
+    def __init__(self, model: str = "gpt-4-turbo-preview", max_tokens: int = 2000, temperature: float = 0.3):
+        self.api_key = os.getenv('OPENAI_API_KEY')
+        if not self.api_key:
+            raise ValueError("OpenAI API key is required. Set OPENAI_API_KEY environment variable or pass it to constructor.")
+        
+        self.prompts_folder_path = os.getenv('PROMPTS_FOLDER_PATH')
+        self.client = OpenAI(api_key=self.api_key)
+        self.model = model
+        self.max_tokens = max_tokens
+        self.temperature = temperature
+        self.config_folder_path = os.getenv('CONFIG_FOLDER_PATH')
+        self.days_config = self._load_config("days_mapping.json")
+        self.preferences_config = self._load_config("preferences_mapping.json")
+        self.destinations_config = self._load_config("destinations_mapping.json")
+        self._validate_configs()
+        
+    def _validate_configs(self):
+        if not hasattr(self.destinations_config, 'get') or 'ukrainian_destinations' not in self.destinations_config:
```

Comment:
Using hasattr(config, 'get') is misleading since dict always has 'get'; check for keys directly. Check key existence directly for clarity.
```
if 'ukrainian_destinations' not in self.destinations_config:
    raise RuntimeError("destinations_mapping.json must contain 'ukrainian_destinations' section")
```